homeassistant/components/lutron_caseta/device_trigger.py

```diff
@@ -405,7 +405,7 @@ async def async_get_triggers(
     triggers = []
 
     if not (device := get_button_device_by_dr_id(hass, device_id)):
-        raise InvalidDeviceAutomationConfig(f"Device not found: {device_id}")
```

Comment:
Can we still validate its a valid lutron_caseta device id?

Reply:
good idea, I'll look into it

Reply:
async_get_triggers is not validating that the device that is passed to it, is from the same DOMAIN (lutron_caseta) and will raise an error if not.  

I've added the test back in for this.